main.rb

```diff
@@ -8,40 +8,57 @@
 
 # Class to initialize objects of all the classes and run the methods sequentially to run tictactoe
 class TicTacToeMain
-  # Declare the constants and load the module
-  SIZE = 3
-  PLAYER_O = 'OOO'.freeze
-  PLAYER_X = 'XXX'.freeze
   include BoardInitializer
 
   def initialize
+    @size = nil
+    @player_o = 'O'
+    @player_x = 'X'
     @player = 'player_O'
     @board = {}
     @move_place = MovePlacer.new
     @move_validate = MoveValidator.new
     @game = GameChecker.new
   end
 
+  # Get the size for tictactoe
+  def input_size
+    input_size_flag = true
+    while input_size_flag
+      puts 'Enter the size for tictactoe(3<=size<=6)'
+      @size = gets.chomp.to_i
+      input_size_flag = false if @size >= 3 && @size <= 6
+    end
+  end
+
+  # Set the winning pattern for tictactoe
+  def initialize_winning_pattern
+    @player_o *= @size
+    @player_x *= @size
+  end
+
   # Entry method
   def main
     print "\n"
-    @board = init_board(SIZE, @board)
+    input_size
+    initialize_winning_pattern
+    @board = init_board(@size, @board)
```

Comment:
Can we make BoardInitializer (or say Board) a class, and use an object of it. I think that is better.